hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -249,7 +249,7 @@ protected HoodieWriteMetadata<HoodieData<WriteStatus>> executeClustering(HoodieC
     HoodieData<WriteStatus> statuses = updateIndex(writeStatusList, writeMetadata);
     writeMetadata.setWriteStats(statuses.map(WriteStatus::getStat).collectAsList());
     writeMetadata.setPartitionToReplaceFileIds(getPartitionToReplacedFileIds(clusteringPlan, writeMetadata));
-    validateWriteResult(clusteringPlan, writeMetadata);
+    // if we don't cache the write statuses above, validation will call isEmpty which might retrigger the execution again.
```

Comment:
this comment is confusing: so do we cache or not cache?

Reply:
if we remove this call, then the `validateWriteResult()` is not used any more. can we clean it up? i saw a similar code path in `ClusteringCommitSink()`. we need to address it in flink clustering path too

Reply:
sure. will remove the comment. and will check flink code path as well.

Reply:
incase of flink, we are already operation on a List<WriteStatus> and so there is no fixes required in there.

Reply:
@nsivabalan was this addressed?